versioned_hdf5/cytools.pxd

```diff
@@ -0,0 +1,32 @@
+# This file allows cimport'ing the functions and types declared below from other
+# Cython modules
+
+# Centralized definition of hsize_t, in accordance with libhd5:
+# https://github.com/HDFGroup/hdf5/blob/6b43197b0817596f47670c6b55d26ff7f86d6bd9/src/H5public.h#L301
+# Note that it has been defined like this since 2021; before it was defined as unsigned
+# long long, which is *at least* 64 bit wide: https://github.com/HDFGroup/hdf5/pull/709
+#
+# versioned_hdf5 uses the same datatype for indexing as libhdf5. Notably, this differs
+# from numpy's npy_intp (same as Py_ssize_t, ssize_t, and signed long), which is only 32
+# bit wide on 32 bit platforms, to allow indexing datasets with >=2**31 points per
+# axis on disk, as long as you don't load them in memory all at once.
+#
+# Note that hsize_t is unsigned, which can lead to integer underflows. We chose not to
+# use int64_t as it would be ABI compatible, but not API compatible, with libhdf5
+# (in other words, importing the headers would require hacking).
+#
+# h5py gets this wildly wrong, as it defines hsize_t as long long - which is signed and
+# is not guaranteed to be 64 bit wide everywhere, and confusingly also defines hssize_t
+# as signed long long - which is an alias for long long:
+# https://github.com/h5py/h5py/blob/eaa9d93cc7620f3e7d8cff6f3a739b7d9834aef7/h5py/api_types_hdf5.pxd#L21-L22
+
+from cython cimport Py_ssize_t
+from libc.stdint cimport uint64_t as hsize_t
+
+
+cpdef hsize_t stop2count(hsize_t start, hsize_t stop, hsize_t step) noexcept nogil
+cpdef hsize_t count2stop(hsize_t start, hsize_t count, hsize_t step) noexcept nogil
+cpdef Py_ssize_t ceil_a_over_b(Py_ssize_t a, Py_ssize_t b) noexcept nogil
+cpdef Py_ssize_t smallest_step_after(
+    Py_ssize_t x, Py_ssize_t a, Py_ssize_t m
+) noexcept nogil
```

Comment:
These two functions will be changed to hsize_t in a later PR

Reply:
I guess later on when this becomes `hsize_t` we can pin libhdf5 to the appropriate version in `meson.build`?

Reply:
I found out how to `#include "hdf5.h"` in cython: `cdef extern from "hdf5.h"` causes cython to completely disregard whatever you're writing below (yes, really :facepalm: ) and import from the C header instead